File: 2870-longest-alternating-subarray/2870-longest-alternating-subarray.ts
function alternatingSubarray(nums: number[]): number {
    let j: number = 0;
    let res: number = 0;
    for (let i = 1; i < nums.length; ++i) {
        if (nums[i] !== nums[j] + (i - j) % 2)
            j = i - (nums[i - 1] === nums[i] - 1 ? 1 : 0);
        res = Math.max(res, i - j + 1);
    }
    return res === 1 ? -1 : res;
};

File: 2870-longest-alternating-subarray/2870-longest-alternating-subarray.py
class Solution:
    def alternatingSubarray(self, nums: List[int]) -> int:
        res = j = 0
        
        for i in range(1, len(nums)):
            if nums[i] != nums[j] + (i - j) % 2:
                j = i - int(nums[i - 1] == nums[i] - 1)
            res = max(res, i - j + 1)
            
        return -1 if res == 1 else res